vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java

```diff
@@ -217,21 +212,16 @@ public Optional<String> getAriaDescribedBy() {
                 getElement().getProperty("accessibleDescriptionRef"));
     }
 
-    private boolean autoAddedToTheUi;
-
     /**
      * Creates an empty dialog with a Confirm button
      */
     public ConfirmDialog() {
-        getElement().addEventListener("opened-changed", event -> {
-            if (!isOpened()) {
-                setModality(false);
-            }
-            if (autoAddedToTheUi && !isOpened()) {
-                getElement().removeFromParent();
-                autoAddedToTheUi = false;
-            }
-        });
+        // Initialize auto-add behavior
+        new OverlayAutoAddController<>(this, () -> true);
+
+        addConfirmListener(event -> close());
```

Comment:
Are those listener added intentionally? Sounds like a breaking change that every confirm click automatically closes the dialog. 

Example: 
- user clicks confirm
- Application does some expensive task and failed 
- Confirm dialog is closed even tho the task failed and user should try to click confirm again to execute the task again

Reply:
Currently all of those close the dialog already: https://github.com/vaadin/web-components/blob/e76c4576c2d1875e16681677bbf829637bf48e4f/packages/confirm-dialog/src/vaadin-confirm-dialog-mixin.js#L435-L451

But I was thinking about using a different approach as well, it seems better to stick with the `opened-changed` event than make assumptions what individual actions would do.

Reply:
Oh thanks! Didn't know the web component does that already (I'm using my own confirm dialog based on the normal dialog API) 

Interesting.. another thing to keep in mind when using the official confirm dialog.